components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/PasswordGrantHandler.java

```diff
@@ -136,7 +136,10 @@ private boolean validateUserCredentials(OAuth2AccessTokenReqDTO tokenReq) throws
                 throw new IdentityOAuth2Exception("Authentication failed for " + tokenReq.getResourceOwnerUsername());
             }
         } catch (UserStoreException e) {
-            throw new IdentityOAuth2Exception("Error while authenticating user from user store");
+            String message = UserCoreConstants.ErrorCode.USER_IS_LOCKED + " " + e.getMessage();
```

Comment:
This is incorrect to consider that always UserStoreException has thrown when user account is locked. AFAIR, If user account is locked, it will not throw an exception, but mention as authentication failure.